nucliadb_vectors/src/data_point_provider/mod.rs

```diff
@@ -340,7 +354,7 @@ impl Index {
         std::mem::drop(state);
         std::mem::drop(date);
 
-        if matches!(self.merger_status, MergerStatus::Free) && work_stack_len > ALLOWED_BEFORE_MERGE {
+        if work_stack_len > ALLOWED_BEFORE_MERGE {
```

Comment:
Mmm.. this is for the offline scheduler (the one in the thread), I think if any we should increase it since now merges will happen on the committing thread.
TBH, I was going to delete the if entirely but I left it because of the shards that are already in memory.